src/Sarif.Multitool/RewriteCommand.cs

```diff
@@ -18,9 +19,13 @@ public static int Run(RewriteOptions rewriteOptions)
                 
                 SarifLog actualLog = MultitoolFileHelpers.ReadSarifFile<SarifLog>(rewriteOptions.InputFilePath);
 
-                LoggingOptions loggingOptions = rewriteOptions.ConvertToLoggingOptions();
+                OptionallyEmittedData dataToInsert = OptionallyEmittedData.None;
+                if (rewriteOptions.DataToInsert != null)
+                {
+                    Array.ForEach(rewriteOptions.DataToInsert, data => dataToInsert |= data);
+                }
```

Comment:
These lines (22 through 26) are repeated several times. Consider an extension method:
```
public static OptionallyEmittedData ToFlags(this OptionallyEmittedData[] option)
{
    OptionallyEmittedData dataToInsert = OptionallyEmittedData.None;
    if (option != null)
    {
        Array.ForEach(option, data => dataToInsert |= data);
    }

    return dataToInsert;
}
```
... called like this:
```
OptionallyEmittedData dataToInsert = rewriteOptions.DataToInsert.ToFlags();
```

Reply:
great suggestion

---
In reply to: [201187303](https://github.com/Microsoft/sarif-sdk/pull/939#discussion_r201187303) [](ancestors = 201187303)